scripts/exregional_process_bufr.sh

```diff
@@ -0,0 +1,382 @@
+#   minute        : process obs used for this analysis minute (integer)
+#   trange_start  : obs time window start (minutes before analysis time)
+#   trange_end    : obs time window end (minutes after analysis time)
+#   bkversion     : grid type (background will be used in the analysis)
+#                   0 for ARW  (default)
+#                   1 for FV3LAM
+#-----------------------------------------------------------------------
+
+if [[ "$run_lightning" == true ]]; then
+
+  cat << EOF > namelist.lightning
+ &setup
+  analysis_time = ${YYYYMMDDHH},
+  minute=00,
+  trange_start=-10,
+  trange_end=10,
+  grid_type = "${PREDEF_GRID_NAME}",
+  obs_type = "bufr"
+ /
+
+EOF
+
+  #
+  #-----------------------------------------------------------------------
+  #
+  # link/copy executable file to working directory 
+  #
+  #-----------------------------------------------------------------------
+  #
+  exec_fn="process_Lightning.exe"
+  exec_fp="$EXECdir/${exec_fn}"
+  
+  if [ ! -f "${exec_fp}" ]; then
+    print_err_msg_exit "\
+  The executable specified in exec_fp does not exist:
+    exec_fp = \"${exec_fp}\"
+  Build lightning process and rerun."
+  fi
+  #
+  #
+  #-----------------------------------------------------------------------
+  #
+  # Run the process for lightning bufr file 
+  #
+  #-----------------------------------------------------------------------
+  #
+   PREP_STEP
+   eval $RUN_CMD_UTILS ${exec_fp} ${REDIRECT_OUT_ERR} || print_err_msg "\
+        Call to executable to run lightning process returned with nonzero exit code."
+   POST_STEP
+fi
+
+#
+#-----------------------------------------------------------------------
+#-----------------------------------------------------------------------
+#
+# Link to the observation NASA LaRC cloud bufr file
+#
+#-----------------------------------------------------------------------
+
+obs_file=${obspath_tmp}/${obsfileprefix}.t${HH}${SUBH}z.lgycld.tm00.bufr_d
+print_info_msg "$VERBOSE" "obsfile is $obs_file"
+run_cloud=false
+if [ -r "${obs_file}" ]; then
+   cp_vrfy "${obs_file}" "lgycld.bufr_d"
+   run_cloud=true
+else
+   print_info_msg "$VERBOSE" "Warning: ${obs_file} does not exist!"
+fi
+
+#-----------------------------------------------------------------------
+#
+# Build namelist and run executable for NASA LaRC cloud
+#
+#   analysis_time : process obs used for this analysis date (YYYYMMDDHH)
+#   bufrfile      : result BUFR file name
+#   npts_rad      : number of grid point to build search box (integer)
+#   ioption       : interpolation options
+#                   = 1 is nearest neighrhood
+#                   = 2 is median of cloudy fov
+#   bkversion     : grid type (background will be used in the analysis)
+#                   = 0 for ARW  (default)
+#                   = 1 for FV3LAM
+#-----------------------------------------------------------------------
+
+if [ ${PREDEF_GRID_NAME} == "GSD_RAP13km" ]; then
```

Comment:
Looking through PR #540, it doesn't look like you needed to add the `GSD_RAP13km` grid parameters to `ush/predef_grid_params.yaml` in order for you to successfully run the RRFS.  Would it be best to remove the GSD_RAP13km grid (remove lines 237, 238, 239, 240, and 243), or should we begin supporting the `GSD_RAP13km` grid, as you currently have it here and in `ush/predef_grid_params.yaml`?

Reply:
@MichaelLueken Yes, the test case I run is the RRFS CONUS 3km so this grid was not needed for my run. However, RRFS adds in total 3 new predefined grids and they are used in other test cases. So I think it is best to have it even though it is not needed for this particular test case. I can move the addition of the predefined grid to this PR #647 to make reviewing of PRs for tasks easier. PR #647 will contain common variables for tasks, common predef grids etc. What do you think?

Reply:
@danielabdi-noaa Thank you very much for the explanation.  So, it sounds like there will be two more grids that will need to be added, in addition to `GSD_RAP13km`.  Since the idea behind PR #647 is to contain common variables and grids, I think it would be fine to update `ush/predef_grid_params.yaml` in PR #647.  I would suggest, after updating the predef grids, to leave a comment in the PRs that contain the new predef grids pointing to PR #647 so that the reviewers know where they can be found.

Reply:
@danielabdi-noaa Looking through old PRs, I see that @gsketefian opened PR [#458](https://github.com/ufs-community/regional_workflow/pull/458) in the regional_workflow repository that renamed `GSD_RAP13km` to `RRFS_NA_13km`.  I see this grid in `ush/predef_grid_params.yaml`.  Following this, `GSD_RAP13km` should be renamed to `RRFS_NA_13km` in this PR.

Reply:
@MichaelLueken I have replaced `GSD_RAP13km` with `RRFS_NA_13km`. It has some slight differences but given #458, it is best to keep SRW values. I will put the values from RRFS_dev1 for the `RRFS_NA_13km` grid in comments just for future reference in PR #647